website/docs/r/app_service_virtual_network_connection.html.markdown

```diff
@@ -0,0 +1,145 @@
+---
+subcategory: "App Service (Web Apps)"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_app_service_virtual_network_connection"
+sidebar_current: "docs-azurerm-resource-app-service-virtual-network-connection"
+description: |-
+  Manages Gateway Required Virtual Network Connection (Integrate with an App Service).
+
+---
+
+# azurerm_app_service_virtual_network_connection
+
+Manages Gateway Required Virtual Network Connection (Integrate with an App Service).
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example"
+  location = "West Europe"
+}
+
+resource "azurerm_app_service_plan" "example" {
+  name                = "example"
+  location            = "${azurerm_resource_group.example.location}"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+
+  sku {
+    tier = "Standard"
+    size = "S1"
+  }
+}
+
+resource "azurerm_app_service" "example" {
+  name                = "example-appservice"
+  location            = "${azurerm_resource_group.example.location}"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.example.id}"
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "example-virtual-network"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+  location            = "${azurerm_resource_group.example.location}"
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "GatewaySubnet"
+  resource_group_name  = "${azurerm_resource_group.example.name}"
+  virtual_network_name = "${azurerm_virtual_network.example.name}"
+  address_prefix       = "10.0.1.0/24"
+}
+
+resource "azurerm_public_ip" "example" {
+  name                = "example"
+  location            = "${azurerm_resource_group.example.location}"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+
+  allocation_method = "Dynamic"
+}
+
+resource "azurerm_virtual_network_gateway" "example" {
+  name                = "example"
+  location            = "${azurerm_resource_group.example.location}"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+
+  type     = "Vpn"
+  vpn_type = "RouteBased"
+  sku      = "Basic"
+
+  ip_configuration {
+    name                          = "vnetGatewayConfig"
+    public_ip_address_id          = "${azurerm_public_ip.example.id}"
+    private_ip_address_allocation = "Dynamic"
+    subnet_id                     = "${azurerm_subnet.example.id}"
+  }
+
+  vpn_client_configuration {
+    address_space = ["10.2.0.0/24"]
+  }
+}
+
+
+resource "azurerm_app_service_virtual_network_connection" "example" {
+  app_service_name      = "${azurerm_app_service.example.name}"
+  resource_group_name   = "${azurerm_resource_group.example.name}"
+  virtual_network_id    = "${azurerm_virtual_network.example.id}"

```

Comment:
if we make this `azurerm_virtual_network_gateway_id` we won't need the depends on. I presume you can't create this resource if it doesn't exist then?